### ISSUE TITLE
Tests are ALWAYS running in parallel despite setting a flowsOrder configuration

### ISSUE THREAD

**Issue by joshuadeguzman** [mobile-dev-inc/maestro#1700]
**Describe the bug**
Tests are always running in parallel despite setting the `flowsOrder`, which is a bit confusing because [this doc](https://maestro.mobile.dev/cli/test-suites-and-reports#sequential-execution) clearly outlines that other tests should run right after Flow A, without adding them explicitly in `flowsOrder`.

What happens is that, the rest of the flows not added explicitly, and in order in `flowsOrder` are opening the app in a logged out state. Flow A represents a login existing user flow in our project.

**To Reproduce**
1. Set the `executionOrder` to:

```
executionOrder:
  continueOnFailure: false
  flowsOrder:
    - Flow A
```

2. Run `maestro cloud`

**Expected behavior**
- Flow A should run first before everything else

**Environment information (please complete the following information):**

- Maestro version: 1.35.0
- Platform: Maestro Cloud
- Framework: Flutter
- Device model and OS version: Android

**Workspace (if applicable)**
You can browse our files [here](https://console.mobile.dev/uploads/07274cff-9bdd-47a3-80c3-d5317e7aa25c?teamId=995a357a-9bc4-45e9-8f27-3a73a7a4a6cc&appId=34010645-a558-4bf3-9e07-daf88323fb6d&analysisId=91241f50-ea84-431f-ba35-3238fa4cc95e), needs authentication.

**Comment by OxalcDev** [comment]
Hi,
I have the same issue in test suite execution.
Regardless of what I set in flowsOrder/config.yaml, the test are starting randomly.
I checked the documentation and have already wasted hours of testing scenarios, changing folders, names...etc..

Example:
executionOrder:
  flowsOrder:
    - start_a
    - start_b
    - start_c
    - logintab/login_a
    - logintab/login_b
    
    Same happens if I use 
    - logina
    - loginb
    
test suite execution oder is b or c first.

Is this Bug still open?
If there is a workaround please post the link.

Thx.

**Comment by AntoC-dev** [comment]
Hi,
I have the same issue on v2.0.8. I have defined a list of flows and an exeuction order but it runs with a complete different order:

```
flows:
  - directory1/*
  - directory2/*
executionOrder:
  flowsOrder:
    - directory1/flow1
    - directory1/flow2
    - directory1/flow3
    - directory2/flow1
    - directory2/flow2
```

Could a fix be provided somewhere soon ? Or at least writing this in the [known issues](https://docs.maestro.dev/troubleshooting/known-issues) so that we don't waste a ton of time looking in documentation o understand what we did wrong.

The only workaround I may have found is to name the flows and directories so that it follow the order I want. Then, I just tell in the config:
```
flows:
  - directory1/*
  - directory2/*

local:
  deterministicOrder: true
```
and it works for local

**Comment by Fishbowler** [comment]
@Anto-dev-perso I don't know this area well, since I hate deterministic ordered tests, but I seem to recall that flowsOrder doesn't take a path, but a flow name. Perhaps the filename without the extension? Or the value of the `name` if that's specified? You may need to track this down in source code for an authoritative answer, but I'm pretty certain path isn't it. HTH.

**Comment by AntoC-dev** [comment]
Hi @Fishbowler 

Thanks for the fast response.

I saw indeed that `flowsOrder` take a name so I put the flow without extension. 
Note that all flows have unique names and don't rely on app state from other tests.

I have a `name` attribute define in all my tests. So, I don't know really what to enter in my config to explicitely set an order:
* Should I enter the file name but without the path (in my example, no `directoryX/`) ?
* Should I copy/paste the string value I entered in the `name` part of each flow ?

I couldn't find the answer in [the documentation](https://docs.maestro.dev/cli/test-suites-and-reports#sequential-execution).

**Comment by Fishbowler** [comment]
As per my last response, I don't know. I'm working from memory. I'd delete this feature if I were allowed.

If you find out from source code or from experimentation, let me know and I'll update docs.

**Comment by AntoC-dev** [comment]
From experimentations I did today, I found that putting name of the file (without extension) isn't working.

But putting the name of the flow as described in the flow file `name: This is the flow name` did the trick.

So the final result looks like:
```
flows:
  - directory1/*
  - directory2/*
executionOrder:
  flowsOrder:
    - "Name of the first flow"
    - "Name of the second flow"
    - "Name of the third flow"
[...]
```

Note that I have a warning with this config: 
```
Requested 1 shards, but it cannot be higher than the number of flows (0). Will use 1 shards instead.
```

Anyway, it doesn't seems like a bug but definitively it's worth an addition in the documentation
@Fishbowler

**Comment by erbud** [comment]
@AntoC-dev, I'm at the same point.

**Comment by Fishbowler** [comment]
I worked with someone in the community to unpick the logic last week.

* If there's a `name`, you specify that.
* If there's not, it's the filename without extension

If someone wants to PR an improvement to this, I'll happily review it - it's clear as mud 😁

**Comment by erbud** [comment]
@Fishbowler, Personally, I see the logic as perfectly correct; what confuses me is the warning message:

> Requested 1 shards, but it cannot be higher than the number of flows (0). Will use 1 shards instead.

 I'm missing something.

**Comment by Fishbowler** [comment]
> cannot be higher than the number of flows (0)

You're trying to run zero flows. Your config is incorrect.

**Comment by erbud** [comment]
> You're trying to run zero flows. Your config is incorrect.

Yes, I understand the config is wrong, but I can't see what's wrong. I'll work on it! I'm missing something...

**Comment by Fishbowler** [comment]
If you'd like to work together on debugging stuff like this, the place is our community Slack - https://slack.maestro.dev

**Comment by Fishbowler** [comment]
Cloud wasn't running `flowsOrder` configs correctly on Maestro Cloud - this was fixed last month.
